src/OpenAL/OpenToolkit.OpenAL/ALC/ALContextAttributes.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenToolkit.Audio.OpenAL
+{
+    public class ALContextAttributes
+    {
+        public int? Frequency { get; set; }
+
+        public int? MonoSources { get; set; }
+
+        public int? StereoSources { get; set; }
+
+        public int? Refresh { get; set; }
+
+        public bool? Sync { get; set; }
+
+        public int[] AdditionalAttributes { get; set; }
+
+        public ALContextAttributes()
+        {
+        }
+
+        public ALContextAttributes(int? freq, int? monoSources, int? stereoSources, int? refresh, bool? sync)
+        {
+            Frequency = freq;
+            MonoSources = monoSources;
+            StereoSources = stereoSources;
+            Refresh = refresh;
+            Sync = sync;
+        }
+
+        /// <summary>
+        /// Converts these context attributes to a <see cref="ALC.CreateContext(ALDevice, Span{int})"/> compatible list.
+        /// </summary>
+        /// <returns>The attibute list in the form of a span.</returns>
+        public Span<int> CreateAttributeList()
```

Comment:
Hmmmmm `CreateAttributesList` suggests a List<T> of some kind is returned, but it returns a Span, which is not technically a list.

What about `CreateAttributesAsSpan()`?

Reply:
This is changed to return an `int[]` and is renamed to `CreateAttributeArray` in the latest.